src/erlcloud_ddb_impl.erl

```diff
@@ -240,11 +240,19 @@ client_error(Body, DDBError) ->
                         [_, Type] when
                               Type =:= <<"ProvisionedThroughputExceededException">> orelse
                               Type =:= <<"ThrottlingException">> ->
-                            DDBError#ddb2_error{error_type = ddb, 
+                            DDBError#ddb2_error{error_type = ddb,
                                                 should_retry = true,
                                                 reason = {Type, Message}};
+                        [_, Type] when
+                              Type =:= <<"TransactionCanceledException">> ->
+                            CancellationReasons0 = proplists:get_value(<<"CancellationReasons">>, Json, []),
+                            CancellationReasons = [{proplists:get_value(<<"Code">>, R),
+                                                    proplists:get_value(<<"Message">>, R, <<>>)} || R <- CancellationReasons0],
+                            DDBError#ddb2_error{error_type = ddb,
+                                                should_retry = false,
```

Comment:
why not retry if one of the reason for transaction fails is conflicting transactions?
@hmartel-alertlogic FYI

Reply:
@motobob I don't think it's an OK idea. Transactions may be canceled for reasons ranging from provisioned throughput exceptions to conditional check failure.

If we want to really "magic" it, we could make a `should_retry(CancellationReasonsCodes)` that would *only* set retry to true if the CancellationReasons are all `None`,  `ProvisionedThroughputExceeded`, or `ThrottlingError`. @hmartel-alertlogic what do you think?

Reply:
@motobob FYI, check out [TransactWriteItems](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_TransactWriteItems.html) docs, there's `ConditionalCheckFailed`, `ItemCollectionSizeLimitExceeded`, `TransactionConflict`, and `ValidationError`. *None* of these should ever be retried on.

Reply:
As suggested by @nalundgaard, I added a function to set `should_retry` to `true` only if cancellation reasons are either: `None`, `ThrottlingError` and/or `ProvisionedThroughputExceeded`.

Reply:
Retrying in case of conflicting transaction would break patterns such as:

- read some values
- prepare update using read values
- update only if read values have not changed

The "prepare update using read values" needs to be redone. We would need to pass a fun to do that work... goes way beyond what erlclould_ddb2 should do.

Reply:
RIght, @hmartel-alertlogic auto-retrying on a conflicting transaction is a non-starter, but the question is whether auto-retrying on exclusively throttling/provisioned throughput exceeded errors is a good idea. That is what @va17 has implemented here. Let me know if you think that is an issue.